app/cli.py

```diff
@@ -0,0 +1,239 @@
+
+
+def create_install_config_file(clusters, base_install_config):
+    for _cluster in clusters:
+        base_install_config_copy = copy.deepcopy(base_install_config)
+        cluster_name = _cluster["name"]
+        install_dir = _cluster["install-dir"]
+        base_install_config_copy["metadata"]["name"] = cluster_name
+        base_install_config_copy["baseDomain"] = _cluster["baseDomain"]
+        base_install_config_copy["platform"] = {
+            _cluster["platform"]: {"region": _cluster["region"]}
+        }
+        Path(install_dir).mkdir(parents=True, exist_ok=True)
+        with open(os.path.join(install_dir, "install-config.yaml"), "w") as fd:
+            fd.write(yaml.dump(base_install_config_copy))
+
+    return clusters
+
+
+def get_pull_secret_data(pull_secret_file):
+    with open(pull_secret_file) as fd:
+        return json.load(fd)
+
+
+def get_base_install_config_data():
+    with open("app/manifests/install-config.yaml") as fd:
+        return yaml.safe_load(fd)
+
+
+def set_debug_os_flags():
+    os.environ["OCM_PYTHON_WRAPPER_LOG_LEVEL"] = "DEBUG"
+    os.environ["OPENSHIFT_PYTHON_WRAPPER_LOG_LEVEL"] = "DEBUG"
+
+
+def get_local_ssh_key():
+    with open(os.path.expanduser("~/.ssh/id_rsa.pub")) as fd:
+        return fd.read().strip()
+
+
+@click.command()
+@click.option("-i", "--install", is_flag=True, help="Install Openshift cluster/s")
+@click.option("-u", "--uninstall", is_flag=True, help="Uninstall Openshift cluster/s")
+@click.option(
+    "-p",
+    "--parallel",
+    help="Run clusters install/uninstall in parallel",
+    is_flag=True,
+    show_default=True,
+)
+@click.option(
+    "--clusters-install-data-directory",
+    help="""
+\b
+Path to cluster install data
+    For install this will be used to store the install data.
+    For uninstall this will be used to uninstall the cluster and must be provided.
+""",
+    default=os.environ.get(
+        "CLUSTER_INSTALL_DATA_DIRECTORY",
+        "/openshift-cli-installer/clusters-install-data",
+    ),
+    type=click.Path(exists=True),
+    show_default=True,
+)
+@click.option(
+    "--pull-secret-file",
+    help="Path to pull secret json file, can be obtained from console.redhat.com",
+    required=True,
+    default=os.environ.get("PULL_SECRET"),
+    type=click.Path(exists=True),
+    show_default=True,
+)
+@click.option(
+    "-c",
+    "--cluster",
+    type=DictParamType(),
+    help="""
+\b
+Cluster/s to install.
+Format to pass is:
+    'name=cluster1;baseDomain=aws.domain.com;platform=aws;region=us-east-2;version=4.14.0-ec.2'
+Required parameters:
+    name: Cluster name.
+    baseDomain: Base domain for the cluster.
+    platform: Cloud platform to install the cluster on. (Currently only AWS supported).
+    region: Region to use for the cloud platform.
+    version: Openshift cluster version to install
+
+Every parameter in install-config.yaml can be override by the user.
+For example:
+    fips=true
+    compute_platform_aws_type=m5.xlarge
```

Comment:
maybe we can add example for how to set number of machines?

Reply:
Done